src/main/kotlin/graphics/scenery/volumes/Volume.kt

```diff
@@ -963,32 +1014,92 @@ open class Volume(
+        }
+
+        /**
+         * Reads raw volumetric data from a [file], splits it into buffers of at most, and as close as possible to,
+         * [sizeLimit] bytes and creates a volume from each buffer.
+         *
+         * Returns the list of volumes.
+         */
+        @JvmStatic
+        fun <T: RealType<T>> fromPathRawSplit(
+            file: Path,
+            type: T,
+            sizeLimit: Long = 2000000000L,
+            hub: Hub
+        ): Pair<Node, List<Volume>> {
+
+            val infoFile = file.resolveSibling("stacks.info")
+
+            val lines = Files.lines(infoFile).toList()
+
+            logger.debug("reading stacks.info (${lines.joinToString()}) (${lines.size} lines)")
+            val dimensions = Vector3i(lines.get(0).split(",").map { it.toInt() }.toIntArray())
+            val bytesPerVoxel = type.bitsPerPixel/8
+
+            var slicesRemaining = dimensions.z
+            var bytesRead = 0L
+            var numPartitions = 0
 
-                    imageData.flip()
-                    imageData
+            val slicesPerPartition = floor(sizeLimit.toFloat()/(bytesPerVoxel * dimensions.x * dimensions.y)).toInt()
+
+            val children = ArrayList<Volume>()
+
+            while (slicesRemaining > 0) {
+                val slices = if(slicesRemaining > slicesPerPartition) {
+                    slicesPerPartition
+                } else {
+                    slicesRemaining
                 }
 
-                volumes.add(BufferedVolume.Timepoint(id, buffer))
+                val partitionDims = Vector3i(dimensions.x, dimensions.y, slices)
+                val size = bytesPerVoxel * dimensions.x * dimensions.y * slices
+
+                val window = bytesRead to bytesRead+size-1
+
+                logger.debug("Reading raw file with offsets: $window")
+                val buffer = readRawFile(file, partitionDims, bytesPerVoxel, window)
+
+                val volume = ArrayList<BufferedVolume.Timepoint>()
+                volume.add(BufferedVolume.Timepoint(file.fileName.toString(), buffer))
+                children.add(fromBuffer(volume, partitionDims.x, partitionDims.y, partitionDims.z, type, hub))
+
+                slicesRemaining -= slices
+                numPartitions += 1
+                bytesRead += size
             }
 
-            return fromBuffer(volumes, dimensions.x, dimensions.y, dimensions.z, UnsignedShortType(), hub)
+            val parent = RichNode()
+            children.forEach { parent.addChild(it) }
+
+            return parent to children
+        }
+
+        /**
+         * Positions [volumes] back-to-back without gaps, using their [pixelToWorld] ratio. Can, e.g., be used
+         * with [fromPathRawSplit] to load volume files greater than 2 GiB into sliced partitions and place
+         * the partitions back-to-back, emulating a single large volume in the scene.
+         */
+        fun positionSlices(volumes: List<Volume>, pixelToWorld: Float) {
```

Comment:
I am not a fan of having this as a static method. How about an extension function of `Volume` or `RichNode`, so you can call this directly on the parent node? Also, `pixelToWorld` should get the default value `volumes.first().pixelToWorldRatio`, so you don't always have to hand it over.

Reply:
Sure, moved the `positionSlices` function to `RichNode` where it is called `positionVolumeSlices` and does not take `pixelToWorld` as parameter